### ISSUE TITLE
Incorrect PPD

### ISSUE THREAD

**Issue by mokuyo** [FoldingAtHome/fah-web-client-bastet#90]
Work Unit Details always shows the same PPD whilst in progress. The v7 Web Control gave an estimated PPD that was much more reflective of the actual performance being achieved, which was nice to know.

**Comment by kbernhagen** [comment]
PPD on Apple M1, when available, is always exactly 86400. Which seems suspicious as it is seconds in a day.

**Comment by kbernhagen** [comment]
This is a client issue.

**Comment by mokuyo** [comment]
> PPD on Apple M1, when available, is always exactly 86400. Which seems suspicious as it is seconds in a day.

Interesting. My M1 Max shows 172800 ppd for the current unit, and sometimes shows 259200, both also a multiple of 86400.

**Comment by kbernhagen** [comment]
My intel Mac is also apparently always showing an integer multiple of 86400 for PPD.

**Comment by kbernhagen** [comment]
Not tested, but I suspect fah-client Unit.cpp:256
The lines using wu.data.estimate might indirectly short-circuit the ppd estimate.
```cpp
uint64_t Unit::getRunTimeEstimate() const {
  // If valid, use estimate provided by the WS
  uint64_t estimate = data->selectU64("wu.data.estimate", 0);
  if (estimate) return estimate;

  // Make our own estimate
  if (getKnownProgress() && lastKnownProgressUpdateRunTime)
    return lastKnownProgressUpdateRunTime / getKnownProgress();

  // Make a wild guess based on timeout
  return 0.2 * data->selectU64("assignment.data.timeout", 1);
}
```

**Comment by mokuyo** [comment]
Currently running project 18491, and showing 345600 PPD, another multiple of 86400, using the same hardware. I wish all my WUs could achieve this kind of performance. 😉

**Comment by kbernhagen** [comment]
I think this is fixed as of v8.2.2
